### ISSUE TITLE
ANSI colour escape codes with python-pylint

### ISSUE THREAD

**Issue by owlas** [flycheck/flycheck#1020]
# ANSI colour escape codes with python-pylint

I have just enabled flycheck for the first time, which has detected my python-pylint executable. However, I am getting the following error that seems to arise because ANSI colours are not being escaped properly and are messing up message parsing:

```shell
Suspicious state from syntax checker python-pylint: Checker python-pylint returned non-zero exit code 16, but no errors from output: ^[[7;33m************* Module flycheck_mymodule^[[0m
flycheck_mymodule.py:1:0:^[[1mC^[[0m:^[[1mmissing-docstring^[[0m:^[[1mMissing module docstring^[[0m
```

My setup:
```python
  python-flake8
    - predicate:  nil
    - executable: Not found

  python-pylint
    - predicate:  t
    - executable: Found at /Users/oliverlaslett/miniconda2/bin/pylint

  python-pycompile
    - predicate:  t
    - executable: Found at /Users/oliverlaslett/miniconda2/bin/python
```

```shell
GNU Emacs 24.5.1 (x86_64-apple-darwin15.2.0) of 2015-12-20 on elcapitanvm.local
```

```shell
Flycheck version: 26snapshot (package: 20160419.712)
```

This was reproduced by:
```shell
$ emacs -Q file.py
$ M-x package-initialize
$ M-x global-flycheck-mode
```

**Comment by lunaryorn** [comment]
@owlas Thank you for reporting this issue.

However, Flycheck doesn't "escape" (i.e. parse?) ANSI colour codes, because syntax checkers shouldn't be printing colourised output in the first place.  It simply doesn't make sense to write ANSI colours in non-TTY environments.

Do you happen to have a pylint configuration somewhere that enables coloured output?

**Comment by owlas** [comment]
Interesting. I just `pip install`ed pylint and used it out of the box. I'm using Iterm2 on mac. Could the emulator be formatting the output?

**Comment by owlas** [comment]
Ok, very sorry, your comment has helped me to find the problem.

By testing a file in a new project, I've found that the project (from GH) I was using has a custom `.pylintrc` file that is enabled colour output.

**Comment by owlas** [comment]
Would it be reasonable to make a feature request that colour codes are escaped properly?

**Comment by lunaryorn** [comment]
@owlas I'd not implement such a request myself, and my vote would also go against merging a corresponding pull request.

I think that syntax checkers should never emit these colour codes in the first place, and I'd not like to add complex code—parsing these sequences is not that trivial—to work around what I think is a configuration error or bug in the syntax checker.  No tool should write colour sequences to non-TTY devices unless it's forced to do, and in Flycheck we never force syntax checkers to do that.

It's additional complexity and maintenance effort for a problem that's much better solved in the syntax checker itself.

@cpitclaudel, your opinion?

We could try to force pylint not to use coloured output by adding an additional option to its `:command`, though.  Do you know the corresponding flag?

**Comment by owlas** [comment]
I believe the `no-color` flag is needed to remove the colours.

I think you make a convincing argument. Not to mention the fact that the option to show colours is more a user preference than something that should be committed into a project's `.pylintrc`

**Comment by lunaryorn** [comment]
@owlas I'm afraid `pylint` 1.6.x doesn't know about this option here 😕

**Comment by cpitclaudel** [comment]
> It's additional complexity and maintenance effort for a problem that's much better solved in the syntax checker itself.
> @cpitclaudel, your opinion?

I agree entirely; pylint should not produce these colors. On the other hand, it might not be that hard to remove the sequences: `ansi-color-regexp` probably covers most of them.
